analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -20,543 +20,550 @@
                 }
             }
+        }
 
-            private ISymbol GetDeclaredSymbol(SyntaxNode syntaxNode) =>
-                getSemanticModel(syntaxNode).GetDeclaredSymbol(syntaxNode);
+        private static bool IsEmptyConstructor(BaseMethodDeclarationSyntax constructorDeclaration) =>
+            !constructorDeclaration.HasBodyOrExpressionBody()
+            || (constructorDeclaration.Body is { } && constructorDeclaration.Body.Statements.Count == 0);
 
-            private void StoreRemovableVariableDeclarations(BaseFieldDeclarationSyntax node)
-            {
-                foreach (var variable in node.Declaration.Variables)
-                {
-                    var symbol = GetDeclaredSymbol(variable);
-                    if (IsRemovableMember(symbol))
-                    {
-                        PrivateSymbols.Add(symbol);
-                        FieldLikeSymbols.Add(symbol, variable);
-                    }
-                }
-            }
+        private static bool IsDeclaredInPartialClass(ISymbol methodSymbol)
+        {
+            return methodSymbol.DeclaringSyntaxReferences
+                .Select(GetContainingTypeDeclaration)
+                .Any(IsPartial);
 
-            private static bool IsEmptyConstructor(BaseMethodDeclarationSyntax constructorDeclaration) =>
-                !constructorDeclaration.HasBodyOrExpressionBody()
-                || (constructorDeclaration.Body is { } && constructorDeclaration.Body.Statements.Count == 0);
+            static TypeDeclarationSyntax GetContainingTypeDeclaration(SyntaxReference syntaxReference) =>
+                syntaxReference.GetSyntax().FirstAncestorOrSelf<TypeDeclarationSyntax>();
 
-            private static bool IsDeclaredInPartialClass(ISymbol methodSymbol)
+            static bool IsPartial(TypeDeclarationSyntax typeDeclaration) =>
+                typeDeclaration.Modifiers.AnyOfKind(SyntaxKind.PartialKeyword);
+        }
+
+        private static bool IsRemovableMethod(IMethodSymbol methodSymbol) =>
+            IsRemovableMember(methodSymbol)
+            && (methodSymbol.MethodKind is MethodKind.Ordinary or MethodKind.Constructor or MethodKindEx.LocalFunction)
+            && !methodSymbol.IsMainMethod()
+            && (!methodSymbol.IsEventHandler() || !IsDeclaredInPartialClass(methodSymbol)) // Event handlers could be added in XAML and no method reference will be generated in the .g.cs file.
+            && !methodSymbol.IsSerializationConstructor()
+            && !methodSymbol.IsRecordPrintMembers();
+
+        private static bool IsRemovable(ISymbol symbol) =>
+            symbol is { IsImplicitlyDeclared: false, IsVirtual: false }
+            && !HasAttributes(symbol)
+            && !symbol.IsSerializableMember()
+            && !symbol.ContainingType.IsInterface()
+            && symbol.GetInterfaceMember() is null
+            && symbol.GetOverriddenMember() is null;
+
+        private static bool HasAttributes(ISymbol symbol)
```

Comment:
This is the important part.
I check for the getter/setter and the property itself for attributes in the special case.